htdocs/core/lib/ticket.lib.php

```diff
@@ -867,11 +820,73 @@ function show_ticket_messaging($conf, $langs, $db, $filterobj, $objcon = '', $no
 		$out .= "</ul>\n";
 	}
 
-
 	if ($noprint) return $out;
 	else print $out;
 }
 
+/**
+ * Get timeline icon
+ * @param ActionComm $actionstatic actioncomm
+ * @param array $histo histo
+ * @param int $key key
+ * @return string
+ */
+function getTicketTimelineIcon($actionstatic, $histo, $key)
```

Comment:
May be we should a a & to have variable passed by reference ($histo may be a very large array)

function getTicketTimelineIcon($actionstatic, &$histo, $key)